spark-indexer/src/main/java/org/uniprot/store/spark/indexer/proteome/converter/DatasetProteomeEntryConverter.java

```diff
@@ -25,20 +42,274 @@ public class DatasetProteomeEntryConverter implements Function<Row, ProteomeEntr
-            if (isReference) {
-                type = ProteomeType.REFERENCE;
-            }
+        // upid
+        if (hasFieldName(UPID, row)) {
+            builder.proteomeId(row.getString(row.fieldIndex(UPID)));
+        }
+        //taxonomy
+        if (hasFieldName(TAXONOMY, row)) {
+            Taxonomy taxonomy = new TaxonomyBuilder().taxonId(row.getInt(row.fieldIndex(TAXONOMY))).build();
+            builder.taxonomy(taxonomy);
+        }
+        //strain
+        if (hasFieldName(STRAIN, row)) {
+            builder.strain(row.getString(row.fieldIndex(STRAIN)));
+        }
+        //strain
+        if (hasFieldName(DESCRIPTION, row)) {
+            builder.description(row.getString(row.fieldIndex(DESCRIPTION)));
+        }
+        //modified
+        if (hasFieldName(MODIFIED, row)) {
+            DateTimeFormatter formatter = DateTimeFormatter.ofPattern("yyyy-MM-dd");
+            builder.modified(LocalDate.parse(row.getString(row.fieldIndex(MODIFIED)), formatter));
         }
-        builder.proteomeType(type);
+        // proteome type
+        ProteomeType proteomeType = ProteomeType.NORMAL;
+        boolean isReference = false;
+        boolean isRepresentative = false;
+        if (hasFieldName(IS_REFERENCE_PROTEOME, row)) {
+            isReference = row.getBoolean(row.fieldIndex(IS_REFERENCE_PROTEOME));
+        }
+        if (hasFieldName(IS_REPRESENTATIVE_PROTEOME, row)) {
+            isRepresentative = row.getBoolean(row.fieldIndex(IS_REPRESENTATIVE_PROTEOME));
+        }
+        if (isReference && isRepresentative) {
+            proteomeType = ProteomeType.REFERENCE_AND_REPRESENTATIVE;
+        } else if (isReference) {
+            proteomeType = ProteomeType.REFERENCE;
+        } else if (isRepresentative) {
+            proteomeType = ProteomeType.REPRESENTATIVE;
+        }
+        builder.proteomeType(proteomeType);
+        //genome annotation
+        if (hasFieldName(GENOME_ANNOTATION, row)) {
+            GenomeAnnotation genomeAnnotation = getGenomeAnnotation((Row) row.get(row.fieldIndex(GENOME_ANNOTATION)));
+            builder.genomeAnnotation(genomeAnnotation);
+        }
+        //genome assembly
+        if (hasFieldName(GENOME_ASSEMBLY, row)) {
+            GenomeAssembly genomeAssembly = getGenomeAssembly((Row) row.get(row.fieldIndex(GENOME_ASSEMBLY)));
+            builder.genomeAssembly(genomeAssembly);
+        }
+        //annotation score
+        if (hasFieldName(ANNOTATION_SCORE, row)) {
+            Integer annotationScore = getAnnotationScore((Row) row.get(row.fieldIndex(ANNOTATION_SCORE)));
+            builder.annotationScore(annotationScore);
+        }
+        //component set
+        if (hasFieldName(COMPONENT, row)) {
+            List<Row> componentRows = row.getList(row.fieldIndex(COMPONENT));
+            builder.componentsSet(componentRows.stream().map(this::getComponent).collect(Collectors.toList()));
+        }
```

Comment:
what if this::getComponent returns null? Same comment for this::getCitation call.

Reply:
according to xsd it wont happen